### ISSUE TITLE
Unable to add activity

### ISSUE THREAD

**Issue by okuka** [catalyst/moodle-mod_facetoface#5]
Hi!

We're trying to use Face-to-Face to allow teachers to book courses for their classes at an open event of our university (which happens every two years). We used the plugin the last few times and were quite happy with it. So were quite thrilled to see an update coming around the corner just in time. But unfortunately we can't get it to work and hope you can help us.

Every time we add the activity, the form shows up, we can fill it, but after we hit submit, we only get a blank page. Upon returning, no new activity has been added.

Digging a bit in, all facetoface tables remain completely empty and the error log says:

Potential coding error - active database transaction detected during request shutdown:\n\* line 105 of /course/modlib.php: call to moodle_database->start_delegated_transaction()\n\* line 266 of /course/modedit.php: call to add_moduleinfo()\n, referer:http://moodle.my/course/modedit.php?add=facetoface&type=&course=3&section=2&return=0&sr=0

Our system is:
- mod_facetoface 2.8.0 (Build: 2015021300)
- Moodle 2.8.3+ (Build: 20150205)
- MySQL 5.5.41
- PHP 5.4.36-0+deb7u3
- Apache 2.2.22 (Debian)

We'd appreciate any hints on what we might have to change (or of course a patch).

Thanks in advance.

Oliver

**Comment by lamiette** [comment]
Hi,

Thank you for reporting this. Yes, it does seem to be a coding error
issue. I'll look into getting this resolved for you ASAP and pushing a
patch!

Stacey

**Comment by lamiette** [comment]
Hi Oliver,

I've pushed a patch which hopefully amends this. Let me know if there are still issues related to it.

Thanks,
Stacey

**Comment by okuka** [comment]
Hi Stacy,

thanks for the patch, that enabled the install :-) .
It is working now for the most part, but produces two minor errors.

When signing up to a session, it displays the error:
> Invalid event class name 'mod_facetoface\event\signup', there must be at least one underscore separating object and action words.

I've hunted it down to the `signup.php`, line 139:
> $event = \mod_facetoface\event\signup::create($params);

But since I am aware, that the event will be used at other places, I didn't try much to change it.

And the other small error is, when canceling a booking, it shows the error:
> context (or contextid) is a required event property, system context may be hardcoded in init() method.

But that is about the end of my skills. I can code, but I never did for moodle, so I would need to learn how to write plugins for it.

Those two errors don't seem to interfere with the booking itself, since people can book, cancel, get mails and it appears to work fine so far, except for those errors.

Would be really nice if you could help us out there.
To make up for the hassle we create, I will provide you with a German translation next week.

Thanks in advance and wishing you a nice weekend!
Oliver

PS: For some reason my foot hurts just looking at your profile picture. Must've been a scary halloween costume ;) .

**Comment by digidoug** [comment]
I'm running into the same issue: "Invalid event class name 'mod_facetoface\event\signup'", but in the 2.7 version.  Also have an error on line 80 of classes/observer.php saying unknown function to

 facetoface_instance_created($event->get_context(), $forum);

Commenting this out solves the problem, but not sure why - or what impact it has.

Even with the event/signup error, the record is created and the student is able to register.

Hope that helps,
Doug

**Comment by venumV** [comment]
Same here. I tried to add the Face to Face activity to one of my courses and I got this error, "Call to undefined function facetoface_instance_created()", while executing the facetoface_instance_created()  function in observer.php. Would appreciate if this get fixed soon.

**Comment by lamiette** [comment]
Hi all,

Thanks for reporting these and apologies for the errors! I will try and have these resolved as soon as possible and hopefully by the end of the week.

Thanks for your patience! Just juggling a few things at the moment :)

**Comment by adigitalife** [comment]
We're getting the same error as mention by Oliver above when students try to signup to a session:

Invalid event class name 'mod_facetoface\event\signup', there must be at least one underscore separating object and action words.

Hopefully this gets fixed soon.

Thanks,
Aylwin

**Comment by pnbowen** [comment]
Found the issue, well, 2 issues actually.

The first - an event cannot be just a file - so the event signup is invalid. I changed the filename in the event folder to signup_success.php, as well as the event name, and finally the line in signup.php around 140 -         $event = \mod_facetoface\event\signup_success::create($params);

The second - I don't think $contextmodule is defined (although I could be wrong, since this was my first thing I fixed)
So in both signup.php and cancelsignup.php I added 
$contextmodule = context_module::instance($cm->id);
to just below 
$context = context_course::instance($course->id);

I hope this helps.

Cheers
Peter
